### PR TITLE
Refactor deploying docs

### LINKED ISSUE #8348: Promote setup documentation in Docs TOC
I suspect that the deploy page, https://docs.dask.org/en/latest/how-to/deploy-dask-clusters.html#, is of high enough importance that it should maybe get a top-level listing on the sidebar TOC.  Right now it's in the "How to" section at the same level as "How to set up prometheus", which seems less relevant.

### PR DESCRIPTION
**Pull request by jacobtomlinson**
I've added a top level "Fundamentals" heading for "Deploying Clusters" as things seemed a bit buried under "How To". I've also removed a bunch of things that were duplicated from the `scheduling.rst` page and tried to shift the wording towards deployment rather than task scheduling.

cc @jsignell 

Closes #8348

### CODE REVIEW

**Comment by jsignell** [comment]
Thanks for doing this @jacobtomlinson! I think we need a bunch of top-level redirects unfortunately. There is a whole section in docs/source/conf.py

File: docs/source/deploying-cli.rst [hunk 1]
```diff
@@ -2,10 +2,11 @@ Command Line
 ============
 
 This is the most fundamental way to deploy Dask on multiple machines.  In
-production environments, this process is often automated by some other resource
+production environments this process is often automated by some other resource
 manager. Hence, it is rare that people need to follow these instructions
-explicitly.  Instead, these instructions are useful for IT professionals who
-may want to set up automated services to deploy Dask within their institution.
+explicitly.  Instead, these instructions are useful to helm understand what
```
Comment (jsignell) [inline docs/source/deploying-cli.rst:7]:
```suggestion
explicitly.  Instead, these instructions are useful to understand what
```

File: docs/source/deploying-cloud.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
+Cloud Deployments
```
Comment (jsignell) [inline docs/source/deploying-cloud.rst:1]: I think this should maybe just be called "Cloud" given that it's already under the "Deploying Clusters" section

**Comment by jacobtomlinson** [comment]
Thanks for the review @jsignell. I've added the redirects and addressed the feedback.

**Review by jsignell — APPROVED** [review]
Thanks for this @jacobtomlinson

**Pull request merged**